packages/lib/server/repository/organization.ts

```diff
@@ -0,0 +1,68 @@
+import { IS_TEAM_BILLING_ENABLED } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+import { safeStringify } from "@calcom/lib/safeStringify";
+import { prisma } from "@calcom/prisma";
+import { MembershipRole } from "@calcom/prisma/enums";
+
+import { createAProfileForAnExistingUser } from "../../createAProfileForAnExistingUser";
+
+export class OrganizationRepository {
+  static async createWithOwner({
+    orgData,
+    owner,
+  }: {
+    orgData: {
+      name: string;
+      slug: string;
+      isOrganizationConfigured: boolean;
+      autoAcceptEmail: string;
+      seats: number | null;
+      pricePerSeat: number | null;
+    };
+    owner: {
+      id: number;
+      email: string;
+      nonOrgUsername: string;
+    };
+  }) {
+    logger.debug("Creating organization with owner", safeStringify({ orgData, owner }));
+    const organization = await prisma.team.create({
+      data: {
+        name: orgData.name,
+        isOrganization: true,
+        ...(!IS_TEAM_BILLING_ENABLED ? { slug: orgData.slug } : {}),
+        organizationSettings: {
+          create: {
+            isOrganizationVerified: true,
+            isOrganizationConfigured: orgData.isOrganizationConfigured,
+            orgAutoAcceptEmail: orgData.autoAcceptEmail,
+          },
+        },
+        metadata: {
+          ...(IS_TEAM_BILLING_ENABLED ? { requestedSlug: orgData.slug } : {}),
+          orgSeats: orgData.seats,
+          orgPricePerSeat: orgData.pricePerSeat,
```

Comment:
Would it not make sense to put these on orgSettings now its fully merged? As theyre pretty specific to orgs and not teams

Reply:
I am not too sure on this. This data is pretty temporary as it's needed just for the time the org is paid for. Once an org is paid for there is no use for this.

Reply:
Ahh i see yeah youre right no worries at all then